contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -657,15 +658,19 @@ pltsql_pre_parse_analyze(ParseState *pstate, RawStmt *parseTree)
 					rel_schema = get_authid_user_ext_schema_name(get_cur_db_name(), GetUserNameFromId(GetUserId(), false));
 				}
 
-				if (trig_schema != NULL && strncmp(trig_schema, rel_schema, Min(strlen(rel_schema), strlen(trig_schema))) != 0)
+				if (trig_schema != NULL)
 				{
-					ereport(ERROR,
-							(errcode(ERRCODE_INTERNAL_ERROR),
-							errmsg("Trigger %s.%s on table %s.%s does not exists or table %s.%s does not exists",
-									trig_schema, cmd->name, rel_schema, atstmt->relation->relname, rel_schema, atstmt->relation->relname)));
+					if ((strlen(rel_schema) != strlen(trig_schema)) || strncmp(trig_schema, rel_schema, strlen(rel_schema)) != 0)
```

Comment:
why not use strcmp? there is no specific advantage of using strncmp here as strlen will run into same problems faced by strcmp

Reply:
used strcmp